src/application/services/useMarketplace.ts

```diff
@@ -1,34 +1,58 @@
 import type EditorTool from '@/domain/entities/EditorTool';
-import { type Ref, ref, onMounted } from 'vue';
+import { type Ref, ref, onMounted, watch } from 'vue';
 import { marketplaceService } from '@/domain';
+import type { EditorToolWithUserBinding } from '@/domain/entities/EditorTool';
+import { useAppState } from './useAppState';
 
 /**
  * Composable for the application state
  */
 interface UseMarketplaceComposable {
   /**
-   * All editor tools that are used in notes creation
+   * Get list of all tools with user binding
    */
-  tools: Ref<EditorTool[]>;
+  tools: Ref<EditorToolWithUserBinding[]>
 }
 
 /**
  * Application service for working with the Editor Tools
  */
 export default function (): UseMarketplaceComposable {
   /**
-   *  All editor tools
+   *  List of tools with information, if they are installed by the user
    */
-  const tools = ref<EditorTool[]>([]);
+  const toolsWithUserBindings = ref<EditorToolWithUserBinding[]>([]);
+
+  /**
+   * User tools
+   */
+  const { userEditorTools } = useAppState();
+
+  /**
+   * List of all tools
+   */
+  const availableTools = ref<EditorTool[]>([]);
 
   /**
    * Get list of all tools
    */
   onMounted(async () => {
-    tools.value = await marketplaceService.getAllTools();
+    availableTools.value = await marketplaceService.getAllTools();
+
+    toolsWithUserBindings.value = marketplaceService.getToolsWithUserBindings(userEditorTools.value, availableTools.value);
```

Comment:
I think you don't need this call, just add "[immediate](https://vuejs.org/guide/essentials/watchers.html#eager-watchers): true" to the watcher

Reply:
done